### PR TITLE
Allow documentation comments to function as br

### PR DESCRIPTION
**Pull request by mtdowling**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-model/src/test/resources/software/amazon/smithy/model/loader/valid/expects-br.smithy [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace com.example /// Doc comment on Foo
+
+string Foo /// Doc comment on Bar
+
+string Bar /// Doc comment on Baz
+
+string Baz, /// Doc comment on Bam
+
+string Bam,,,/// Dangling
```
Comment (JordonPhillips) [inline smithy-model/src/test/resources/software/amazon/smithy/model/loader/valid/expects-br.smithy:11]: This example looks to me like a good reason for not making this change. If I were writing a doc comment like this, I would expected it to apply to the shape that it shares a line with

**Review by JordonPhillips — APPROVED** [review]

File: smithy-model/src/test/resources/software/amazon/smithy/model/loader/valid/expects-br.smithy (hunk 1 above)
Comment (mtdowling) [reply smithy-model/src/test/resources/software/amazon/smithy/model/loader/valid/expects-br.smithy:11]: Totally agree. I don't like it at all. It worked before though, and it's allowed in the grammar. I updated the PR to emit a warning for this. I think we should consider removing support in the future after the warning has had time to be seen by modelers. I did see some impacted models inside of Amazon.

**Review by sugmanue — APPROVED** [review]

File: smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlTokenizer.java [hunk 2]
```diff
@@ -615,27 +616,56 @@ public void expectAndSkipWhitespace() {
...
+            case NEWLINE:
+            case COMMENT:
+                clearDocCommentLinesForBr();
+                next();
+                skipWs();
+                break;
+            case DOC_COMMENT:
+                // A doc comment was found at the end of a line. Ignore the comment as it's part of the br and does
+                // not come after the br. The modeler most likely meant to use "//" here.
+                clearDocCommentLinesForBr();
```
[the remark this answers is not in the document]
Comment (mtdowling) [reply smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlTokenizer.java:650]: Good point. I'll make the other message emitted a little broader to cover this case too.

File: smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlTokenizer.java [hunk 3]
```diff
@@ -615,27 +616,56 @@ public void expectAndSkipWhitespace() {
...
-            clearDocCommentLinesForBr();
-            skipWs();
-            if (line == getLine() && hasNext()) {
+        }
+
+        // The following tokens are allowed tokens that contain newlines.
+        switch (getCurrentToken()) {
+            case NEWLINE:
+            case COMMENT:
+                clearDocCommentLinesForBr();
```
[the remark this answers is not in the document]
Comment (mtdowling) [reply smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlTokenizer.java:643]: Yes. For example, using doc comments in a trait value, traling at the end of a shape definition, etc.

**Review by sugmanue — APPROVED** [review]

**Pull request merged**